src/Microsoft.ML.Data/Transforms/TypeConverting.cs

```diff
@@ -274,29 +270,36 @@ private TypeConvertingTransformer(IHost host, ModelLoadContext ctx)
             // *** Binary format ***
             // <base>
             // for each added column
-            //   byte: data kind, with high bit set if there is a key range
-            //   if there is a key range
-            //     ulong: min
-            //     ulong: max (0 for unspecified)
-            //     byte: contiguous
+            //   byte: data kind, with high bit set if there is a keyCount
+            //   if there is a keyCount
+            //     ulong: keyCount (0 for unspecified)
 
             _columns = new ColumnInfo[columnsLength];
             for (int i = 0; i < columnsLength; i++)
             {
                 byte b = ctx.Reader.ReadByte();
                 var kind = (DataKind)(b & 0x7F);
                 Host.CheckDecode(Enum.IsDefined(typeof(DataKind), kind));
-                KeyRange range = null;
+                KeyCount keyCount = null;
                 if ((b & 0x80) != 0)
                 {
-                    range = new KeyRange();
-                    range.Min = ctx.Reader.ReadUInt64();
+                    if (ctx.Header.ModelVerWritten < 0x00010004)
```

Comment:
It would be better to declare a constant for `0x00010004` rather than hard-coding that number everywhere.